root_orchestrator/system-manager-python/services/service_management.py

```diff
@@ -1,38 +1,61 @@
 import logging
 
-from ext_requests.apps_db import (
-    mongo_delete_job,
-    mongo_find_app_by_id,
-    mongo_find_job_by_id,
-    mongo_get_all_jobs,
-    mongo_get_applications_of_user,
-    mongo_get_jobs_of_application,
-    mongo_insert_job,
-    mongo_set_microservice_id,
-    mongo_update_application_microservices,
-    mongo_update_job,
-)
 from ext_requests.net_plugin_requests import net_inform_service_deploy, net_inform_service_undeploy
+from resource_abstractor_client import app_operations, job_operations
 from services.instance_management import request_scale_down_instance
 from sla.versioned_sla_parser import parse_sla_json
 
 
+def insert_job(microservice):
+    logging.log(logging.INFO, "MONGODB - insert job...")
+    # jobname and details generation
+    job_name = (
+        microservice["app_name"]
+        + "."
+        + microservice["app_ns"]
+        + "."
+        + microservice["microservice_name"]
+        + "."
+        + microservice["microservice_namespace"]
+    )
+    microservice["job_name"] = job_name
+    job_content = {
+        "job_name": job_name,
+        **microservice,  # The content of the input file
+    }
+
+    # job insertion
+    new_job = job_operations.create_job(job_content)
+    if new_job is None:
+        logging.error(f"job not inserted - {job_name}")
+        return None
+
+    logging.info("job {} inserted".format(str(new_job.get("_id"))))
+    return str(new_job.get("_id"))
+
+
 def create_services_of_app(username, sla, force=False):
     data = parse_sla_json(sla)
     logging.log(logging.INFO, sla)
     app_id = data.get("applications")[0]["applicationID"]
     last_service_id = ""
-    application = mongo_find_app_by_id(app_id, username)
+    application = app_operations.get_app_by_id(app_id, username)
+
     if application is None:
         return {"message": "application not found"}, 404
     for microservice in data.get("applications")[0].get("microservices"):
         if not valid_service(microservice):
             return {"message": "invalid service name or namespace"}, 403
         # Insert job into database
         service = generate_db_structure(application, microservice)
-        last_service_id = mongo_insert_job(service)
+        last_service_id = insert_job(service)
+        if last_service_id is None:
+            # TODO: what should be done to previously inserted services?
```

Comment:
Thanks for adding this check. 
I think previous services should stay where they are, nothing wrong with them. What should be done instead is:
- logging an error; this way, we trigger an alarm in the root dashboard. If there is an insert failure there is likely an issue in the cluster. 
- return in the response maybe a status code for each one of the services. like
```
{
  "job_id": str(last_service_id)  // leave it for compatibility with current dashboard
  "services":[
  {
       "service_id":id
       "status":int //status code 200/500/...
       "description":string //error description
      }
  ]
} 
```
The current dashboard anyway queries the system, but this might be helpful for come popup errors.

Reply:
Good Idea! I'll implement that.